nautobot_netbox_importer/management/commands/import_netbox_objectchange_json.py

```diff
@@ -0,0 +1,105 @@
+
+from nautobot_netbox_importer.utils import ProgressBar
+from nautobot_netbox_importer.command_utils import validate_netbox_version, initialize_logger
+from nautobot_netbox_importer.diffsync.models.validation import netbox_pk_to_nautobot_pk
+
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    """Implementation of import_netbox_objectchange_json command."""
+
+    help = "Import ObjectChange objects from NetBox YAML data dump into Nautobot's database"
+
+    def add_arguments(self, parser):
+        """Add parser arguments to the import_netbox_objectchange_json management command."""
+        parser.add_argument("json_file", type=argparse.FileType("r"))
+        parser.add_argument("objectchange_json_file", type=argparse.FileType("r"))
+        parser.add_argument("netbox_version", type=version.parse)
+        parser.add_argument("--dryrun", type=bool, default=False)
+
+    def handle(self, *args, **options):
+        """Handle execution of the import_netbox_json management command."""
+
+        def process_objectchange(entry):
+            """Processes on objectchange entry to migrate from Netbox to Nautobot."""
+            try:
+                entry["fields"]["changed_object_type"] = ContentType.objects.filter(
+                    id=nautobot_contenttype_mapping[netbox_contenttype_mapping[entry["fields"]["changed_object_type"]]]
+                ).first()
+            except KeyError:
+                logger.warning(
+                    f'{netbox_contenttype_mapping[entry["fields"]["changed_object_type"]]} key is not mapped'
+                )
+                return
+
+            if entry["fields"]["related_object_type"]:
+                entry["fields"]["related_object_type"] = ContentType.objects.filter(
+                    id=nautobot_contenttype_mapping[netbox_contenttype_mapping[entry["fields"]["related_object_type"]]]
+                ).first()
+            else:
+                del entry["fields"]["related_object_type"]
+
+            modelname = entry["fields"]["changed_object_type"].model
+            entry["fields"]["changed_object_id"] = netbox_pk_to_nautobot_pk(
+                modelname, entry["fields"]["changed_object_id"]
+            )
+            if entry["fields"]["related_object_id"]:
+                modelname = entry["fields"]["related_object_type"].model
+                entry["fields"]["related_object_id"] = netbox_pk_to_nautobot_pk(
+                    modelname, entry["fields"]["related_object_id"]
+                )
+
+            entry["fields"]["user"] = User.objects.filter(username=entry["fields"]["user_name"]).first()
+            if not entry["fields"]["user"]:
+                logger.error(f'Username {entry["fields"]["user_name"]} not present in DB.')
+                return
+
+            if not options["dryrun"]:
+                obj = ObjectChange.objects.create(**entry["fields"])
+                obj.time = entry["fields"]["time"]
+                obj.save()
```

Comment:
Would it be appropriate to call `obj.full_clean()` before saving it?

Reply:
totatlly